### PR TITLE
Add forecast day output for CICE via timestep stream

### PR DESCRIPTION
**Pull request by NickSzapiro-NOAA**
## Commit Queue Requirements:

- [x] Fill out all sections of this template.
- [x] All sub component pull requests have been reviewed by their code managers.
- [x] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [x] Commit 'test_changes.list' from previous step
---
## Description:

To have time-averaged CICE output over a forecast day, we use a timestep stream, like
```
histfreq            = '1'
histfreq_n          = 120
hist_avg            = .true.
```
for a CICE timestep of 720s. The other stream options are calendar-based and "daily" output files start over at 00Z (like calendar days), without the desired time_bounds if the forecast starts after 00Z. As it may be better for workflow not to need extra post-processing steps to have daily forecast output (https://github.com/NOAA-EMC/global-workflow/issues/2674), model outputs this stream directly.

icefields_nml section in tests/parm/ice_in.IN has been modified to 1) add all of the hourly stream variables to the timestep stream as well and 2) remove `blkmask` as not needed and fails decomposition tests by construction

Daily ice history files are added to gfsv17 baselines to maintain testing of a forecast day output stream requested by global-workflow. All non-ice history baselines match b4b. Ice history baselines reproduce themselves.

### Commit Message:

```
* UFSWM - Add forecast day output for CICE via timestep stream
```

### Priority:

* Normal

### UFSWM:

* Closes #https://github.com/ufs-community/ufs-weather-model/issues/2422

### Sub component Pull Requests:

* None

### UFSWM Blocking Dependencies:

* None

---
### Regression Test Changes (Please commit test_changes.list):

* PR Updates/Changes Baselines.

### Input data Changes:

* None.

### Library Changes/Upgrades:

* No Updates
  
---

## Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [ ] Derecho
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- [x] CI
- [ ] opnReqTest (complete task if unnecessary)

### CODE REVIEW

File: tests/tests/cpld_control_gfsv17 [hunk 1]
```diff
@@ -78,6 +78,9 @@ export DNATS=0
 export RESTART_N=3
 export RESTART_INTERVAL="${RESTART_N} -1"
 export OUTPUT_FH='0 21 24'
+#CICE forecast day output stream
```
Comment (junwang-noaa) [inline tests/tests/cpld_control_gfsv17:83]: Would you please run the tests for a couple of days and show the time stamp change with this update?
Also could you please make the changes to other gfsv17 tests too? Thanks

Comment (NickSzapiro-NOAA) [reply tests/tests/cpld_control_gfsv17:83]: Done. On hera, please see 
`ls /scratch1/NCEPDEV/nems/Nick.Szapiro/tasks/ice_24hrOutput/ufs-weather-model/tests/run_dir/cpld*/history/`

For cpld_control_gfsv17, I ran FHMAX=72 and history includes `iceh.2021-03-23-21600.nc iceh.2021-03-24-21600.nc iceh.2021-03-25-21600.nc` , with time_bounds correct in each

cpld_restart_gfsv17 and cpld_debug_gfsv17 do not have this as they run less than 1 day

**Comment by NickSzapiro-NOAA** [comment]
@junwang-noaa Ice history baselines have been added and tests are as expected on hercules. Please let me know if there is anything more to resolve

**Review by junwang-noaa — DISMISSED** [review]

**Comment by NickSzapiro-NOAA** [comment]
@jkbk2004 A new bl_date is needed for this PR as the history/iceh_06h changed in addition to the newly added history/iceh daily history. Apologies for my mistake as I forgot about the changed iceh_06h file

Please let me know if I can help correct. Only the CICE history/ice*nc baseline files change between these and the previous baseline date

**Comment by BrianCurtis-NOAA** [comment]
The following should fix the problem created here:

- copy baselines from 0909 to a new bl date (were we using 1001 prior to the revert?)
- run develop branch on the test_changes.list files and copy those to 0909

**Comment by jkbk2004** [comment]
OK! I will update bl_date to 20241001.

**Comment by FernandoAndrade-NOAA** [comment]
> The following should fix the problem created here:
> 
>     * copy baselines from 0909 to a new bl date (were we using 1001 prior to the revert?)
> 
>     * run develop branch on the test_changes.list files and copy those to 0909

Understood, we'll move current 0909 to 1001 and regenerate the two gfs cases for 0909 off develop

**Comment by FernandoAndrade-NOAA** [comment]
Adding a note that the 0909 bl date in the Hera and Gaea logs are for the BLs that were later moved to 20241001 before 0909 was restored.

**Comment by BrianCurtis-NOAA** [comment]
> Adding a note that the 0909 bl date in the Hera and Gaea logs are for the BLs that were later moved to 20241001 before 0909 was restored.

Could you do a sanity check on Hera with develop branch and the 0909 baselines to make sure you made the correct adjustments?

**Comment by BrianCurtis-NOAA** [comment]
I did a sanity check on WCOSS2 with the 0909->1001 change with the develop branch and the 0909 baselines and all tests that WCOSS2 runs, passed.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by jkbk2004 — APPROVED** [review]

**Comment by FernandoAndrade-NOAA** [comment]
> > Adding a note that the 0909 bl date in the Hera and Gaea logs are for the BLs that were later moved to 20241001 before 0909 was restored.
> 
> Could you do a sanity check on Hera with develop branch and the 0909 baselines to make sure you made the correct adjustments?

Sanity check testing finally finished for the affected tests, yes 0909 should be good to go. I'm going to go ahead and merge this PR.

**Comment by FernandoAndrade-NOAA** [comment]
Derecho will be skipped this PR due to allocation.

**Pull request merged**